Telegram/BotProject/telegram_bot.rb

```diff
@@ -1,106 +1,53 @@
+  attr_accessor :city_default, :weather_message
 
   def conversation(token)
     begin
-      Telegram::Bot::Client.run(token) do |bot|
-        bot.listen do |message|
-          case message.text
-          when MESSAGE_START
-            bot.api.send_message(chat_id: message.chat.id, text: hello_message(message.from.first_name))
-          when MESSAGE_WEATHER
-            bot.api.send_message(chat_id: message.chat.id, text: QUESTION_WHICH_CITY)
-            bot.listen do |message|
-              city = message.text
-              response_json(URL + city + API_KEY)
-              bot.api.send_message(chat_id: message.chat.id, text: @weather_message)
-              break
-            end
-          when MESSAGE_STOP
-            bot.api.send_message(chat_id: message.chat.id, text:  bye_message(message.from.first_name))
-          else
-            bot.api.send_message(chat_id: message.chat.id, text:  MESSAGE_NOT_UNDERSTAND)
+      bot = our_bot(token)
+      bot.listen do |message|
+        case message.text
+        when MESSAGE_START
+          message_send(bot, message, Greeting.new.hello_message(message.from.first_name))
+        when MESSAGE_WEATHER
+          message_send(bot, message, QUESTION_WHICH_CITY)
+          bot.listen do |message|
+            weather_response = WeatherResponse.new
+            weather_response.response_json(URL + message.text + API_KEY)
+            message_send(bot, message, weather_response.weather_message)
+            break
+          end
+        when MESSAGE_STOP
+          message_send(bot, message, Greeting.new.bye_message(message.from.first_name))
+        when MESSAGE_SET
+          message_send(bot, message, QUESTION_WHICH_CITY)
+          bot.listen do |message|
+            @city_default = message.text
+            WeatherNotification.new.weather_notification_create(bot, message, @city_default)
+            break
           end
+        else
+          message_send(bot, message, MESSAGE_NOT_UNDERSTAND)
         end
       end
-    rescue Telegram::Bot::Exceptions::ResponseError => error
-      ERROR_TOKEN_INVALID + error.to_s
     end
+  rescue Telegram::Bot::Exceptions::ResponseError => error
+    ERROR_TOKEN_INVALID + error.to_s
   end
 
-  def hello_message(user_name)
-    HELLO_STR + user_name + HELLO_EMOJI
-  end
-
-  def bye_message(user_name)
-    BYE_STR + user_name
+  def message_send(bot, message, text)
+    bot.api.send_message(chat_id: message.chat.id, text: text)
   end
 
-  def response_json(url)
-    begin
-      uri = URI(url)
-      response = Net::HTTP.get(uri)
-      if response.nil? || response.include?(ERROR_API_KEY)
-        @weather_message = ERROR_WRONG_HTTP_REQUEST
-        return
-      else
-        @response = JSON.parse(response)
-      end
-    rescue StandardError => error
-      @weather_message = ERROR_WRONG_HTTP_REQUEST
-      ERROR_WRONG_HTTP_REQUEST + error.to_s
-      return
+  def our_bot(token)
```

Comment:
you can add this token to CONSTANT and use it all the time, and you do not need to send this parameter (as you always use the same token)